pkg/controller/node/nodecontroller.go

```diff
@@ -565,6 +677,10 @@ func (nc *NodeController) monitorNodeStatus() error {
 			continue
 		}
 		node := nodeCopy.(*v1.Node)
+
+		// taint or clear NoSchedule taint based on condition change, we should do this before tryUpdateNodeStatus
+		nc.addOrRemoveNoScheduleTaintByCondition(node)
```

Comment:
The test should check node's condition instead of the return value, because the nodeController does not use the return value, some case maybe missed.